### ISSUE TITLE
Collect all HolochainError types in an enum

### ISSUE THREAD

**Issue by jost-s** [holochain/holochain-client-js#281]
Errors thrown by the client are of type `HolochainError`. One property is a `name`, which is a string. Instead it would be better to use an enum, so that there are no spelling mistakes when throwing or catching errors, and to have a central location where all possible errors are placed.

AC:
- Create `HolochainErrorKind` enum.
- Modify `HolochainError` to accept a value of the `HolochainErrorKind` enum instead of a name string.
- Updarte all occurrences of `HolochainError`.

**Comment by ThetaSinner** [comment]
I agree this would be good, but the errors from Holochain are structured to several levels of depth and seriously confusing to decode and decipher. I think we should approach this from Holochain first and simplify what you can receive as errors then look at doing this. What do you think?

**Comment by jost-s** [comment]
We can do that first, but the errors in the client here are not directly mapped to Holochain errors or even unrelated to Holochain. There are for example `InvalidTokenError`, `WebsocketClosedError` or `InvalidMessageFormatError`. Those I'd like to create an enum for instead of using strings literals.
